app/scripts/migrations/160.ts

```diff
@@ -206,6 +208,24 @@ function transformState(state: Record<string, unknown>) {
       currentChainId,
     );
 
+    let caip25PermissionCaveatWithCurrentChainIdAndAccountsSet =
+      caip25PermissionCaveatWithCurrentChainIdSet;
+
+    if (subject.permissions[Caip25EndowmentPermissionName]) {
+      const existingCaveat = subject.permissions[
+        Caip25EndowmentPermissionName
+      ]?.caveats?.find((caveat) => caveat.type === Caip25CaveatType);
+      if (existingCaveat && existingCaveat.value) {
+        const alreadyPermissionedAccounts = getEthAccounts(
+          existingCaveat.value as Caip25CaveatValue,
+        );
+        caip25PermissionCaveatWithCurrentChainIdAndAccountsSet = setEthAccounts(
+          caip25PermissionCaveatWithCurrentChainIdSet,
+          alreadyPermissionedAccounts,
+        );
+      }
+    }
```

Comment:
This can be deduplicated using the `getExistingCaip25PermissionCaveat(subject)` above.

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/33111/commits/feb0add4fa00a230e8ab452f08eca238cd3bb2bd